test/csuite/parse_opts/main.c

```diff
@@ -0,0 +1,315 @@
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+  {NULL, NULL, {0}, NULL, NULL, (char *)"my_store", 0, NULL, NULL, false, false, true, true, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+  {NULL, NULL, {0}, NULL, NULL, (char *)"my_store", 0, NULL, NULL, false, false, true, true, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+};
+
+static EXTENDED_OPTS x_expected[] = {
+  {NULL, 0, 0, 0},
+  {NULL, 0, 0, 0},
+  {NULL, 0, 0, 0},
+  {NULL, 0, 0, 0},
+  {NULL, 0, 0, 0},
+
+  {(char *)"string_opt", true, false, 0},
+  {(char *)"string_opt", true, false, 0},
+  {(char *)"string_opt", false, true, 22},
+  {NULL, true, true, 22},
+  {NULL, false, false, 0},
+};
+
+/*
+ * report --
+ *     Show any changed fields in the options.
+ */
+static void
+report(const TEST_OPTS *opts, EXTENDED_OPTS *x_opts)
+{
+#define REPORT_INT(o, field) \
+    if (o->field != 0)       \
+    printf(#field ": %" PRIu64 "\n", (uint64_t)o->field)
+#define REPORT_STR(o, field) \
+    if (o->field != NULL)    \
+    printf(#field ": %s\n", o->field)
+
+    REPORT_STR(opts, home);
+    REPORT_STR(opts, build_dir);
+    REPORT_STR(opts, tiered_storage_source);
+    REPORT_INT(opts, table_type);
+    REPORT_INT(opts, do_data_ops);
+    REPORT_INT(opts, preserve);
+    REPORT_INT(opts, tiered_storage);
+    REPORT_INT(opts, verbose);
+    REPORT_INT(opts, nrecords);
+    REPORT_INT(opts, nops);
+    REPORT_INT(opts, nthreads);
+    REPORT_INT(opts, n_append_threads);
+    REPORT_INT(opts, n_read_threads);
+    REPORT_INT(opts, n_write_threads);
+    REPORT_STR(x_opts, c_option);
+    REPORT_INT(x_opts, d_option);
+    REPORT_INT(x_opts, e_option);
+    REPORT_INT(x_opts, f_option);
+}
+
+/*
+ * check --
+ *     Call testutil_parse_opts and return options.
+ */
+static void
+check(int argc, char *const *argv, TEST_OPTS *opts, EXTENDED_OPTS *x_opts)
+{
+    int ch;
+    const char *prog;
+    const char *x_usage = " [-c string] [-d] [-e] [-f int]";
+
+    memset(opts, 0, sizeof(*opts));
+    memset(x_opts, 0, sizeof(*x_opts));
+
+    /* This may be called multiple times, so reset the __wt_getopt parser. */
+    __wt_optind = 1;
+    __wt_optreset = 1;
+
+    prog = argv[0];
+    if (strchr(prog, '/') != NULL)
+        prog = strchr(prog, '/') + 1;
+
+    if (strcmp(prog, "parse_opts") == 0) {
+        /* Regular test of testutil_parse_opts, using only the options that it provides. */
+
+        testutil_check(testutil_parse_opts(argc, argv, opts));
+    } else {
+        /*
+         * Test of extended parsing, in which we'll parse some options that we know about and rely
+         * on testutil_parse_opt to cover the options it knows about.
+         */
+        testutil_assert(strcmp(prog, "parse_opt") == 0);
+
+        /*
+         * For this part of the testing, we're extending the list of options we're parsing, and
+         * using testutil_parse_opt to parse a subset of the standard options.
```

Comment:
Can you add to the comment something about how `b:P:T:v` are marked special and therefore, not handled in the switch from `wt_getopt`? Basically I think this test is going to be the "how to use it" example so it needs to be well commented about how it does it.